docs/docs-beta/docs/guides/deployment/docker.md

```diff
@@ -1,3 +1,197 @@
 ---
-title: "Deploying to Docker"
+title: "Deploying with Docker"
+description: A guide to deploying Dagster with Docker.
 ---
+
+This guide provides instructions for deploying Dagster using Docker. It's common to use Docker containers to run the Dagster web server, daemon, and code servers. These containers are typically deployed using AWS ECS or another container-based orchestration system.
+
+## What you'll learn
+
+- The different Docker containers that run as part of a Dagster Docker deployment
+- How to define Docker images for each of these containers
+- How to write a docker-compose file that stands up these containers
+
+<details>
+  <summary>Prerequisites</summary>
+- Familiarity with Docker
+</details>
+
+### Understand the Docker containers that run as part of a Dagster Docker deployment
+
+A typical Dagster Docker deployment includes the following containers:
+- A container for the Dagster web server
+- A container for the [Dagster daemon](/deployment/dagster-daemon)
+- A container for each [code location](/concepts/code-locations), which runs the code server for that code location
+- A container for each run
+
+The containers for the web server and daemon typically share the same image. Each code location typically has its own image, and that image is also used for runs launched for that code location.
+
+### Define a Docker image for the Dagster web server and daemon
```

Comment:
1. Should users do the steps in this guide in order?
2. This should be an H2 heading, not H3